### PR TITLE
cmd: add recover command tool

### PR DESCRIPTION
**Pull request by nolouch**
cc @disksing @siddontang

### CODE REVIEW

File: cmd/pd-recover/main.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	bootstrapCmp := clientv3.Compare(clientv3.CreateRevision(clusterRootPath), "=", 0)
+	resp, err := client.Txn(ctx).If(bootstrapCmp).Then(ops...).Commit()
+	if err != nil {
+		exitErr(err)
+	}
+	if !resp.Succeeded {
+		fmt.Println("cluster already bootstrapped")
+		return
+	}
+	fmt.Println("recover success! please restart PD")
```
Comment (disksing) [inline cmd/pd-recover/main.go]: Why do we need to restart PD?

File: cmd/pd-recover/main.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	ops = append(ops, clientv3.OpPut(pdClusterIDPath, string(uint64ToBytes(*clusterID))))
+	// recover alloc_id
+	allocIDPath := path.Join(rootPath, "alloc_id")
+	ops = append(ops, clientv3.OpPut(allocIDPath, string(uint64ToBytes(*allocID))))
+
+	// recover bootstrap
+	// recover meta of cluster
+	clusterMeta := metapb.Cluster{
+		Id:           *clusterID,
+		MaxPeerCount: uint32(3),
```
Comment (disksing) [inline cmd/pd-recover/main.go]: I suppose we need to add it to command line argument.

File: cmd/pd-recover/main.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	ops = append(ops, clientv3.OpPut(raftBootstrapTimeKey, string(timeData)))
+
+	// the new pd cluster should not bootstrapped by tikv
+	bootstrapCmp := clientv3.Compare(clientv3.CreateRevision(clusterRootPath), "=", 0)
+	resp, err := client.Txn(ctx).If(bootstrapCmp).Then(ops...).Commit()
+	if err != nil {
+		exitErr(err)
+	}
+	if !resp.Succeeded {
+		fmt.Println("cluster already bootstrapped")
```
Comment (disksing) [inline cmd/pd-recover/main.go]: Make it more like an error message. Such as "failed to recover: the cluster is already bootstrapped"

File: cmd/pd-recover/main.go (hunk 2 above)
Comment (siddontang) [inline cmd/pd-recover/main.go]: +1

File: cmd/pd-recover/main.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+		MaxPeerCount: uint32(3),
+	}
+	clusterValue, err := clusterMeta.Marshal()
+	if err != nil {
+		exitErr(err)
+	}
+	ops = append(ops, clientv3.OpPut(clusterRootPath, string(clusterValue)))
+
+	// set raft bootstrap time
+	nano := time.Now().UnixNano()
```
Comment (siddontang) [inline cmd/pd-recover/main.go]: strange bootstrap time here...

File: cmd/pd-recover/main.go (hunk 1 above)
Comment (nolouch) [inline cmd/pd-recover/main.go]: for reload cluster meta info.

File: cmd/pd-recover/main.go (hunk 4 above)
Comment (nolouch) [inline cmd/pd-recover/main.go]: maybe we can recover it by dump tikv data later.

File: cmd/pd-recover/main.go [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+)
+
+func exitErr(err error) {
+	fmt.Println(err.Error())
+	os.Exit(1)
+}
+
+func main() {
+	flag.Parse()
+	if *clusterID == 0 {
```
Comment (siddontang) [inline cmd/pd-recover/main.go]: print and exit

**Pull request merged**